### PR TITLE
Orchard note commitment tree test vectors

### PR DESCRIPTION
**Pull request by dconnolly**
## Motivation

We need to check conformance between our implementation of Orchard note and value commitments, note commitment trees, and anything else that uses the Sinsemilla hash.

### Specifications

Fixed a bug about not encoding the *j* in S(*j*) as a 10 bit in 4 little endian bytes, we just were just encoding it with 2 bytes. 

image

https://zips.z.cash/protocol/protocol.pdf#concretesinsemillahash

## Solution

Bring in several test vectors for Orchard note commitment tree empty roots, and Sinsemilla test vectors, from the zcash-hackworks library, and cross-checked against existing implementations in the ECC `orchard` crate.

## Review

@conradoplg @upbqdn (if they have time) 

### Reviewer Checklist

  - [ ] Code implements Specs and Designs
  - [x] Tests for Expected Behaviour

## Follow Up Work

Partially addresses https://github.com/ZcashFoundation/zebra/issues/2079, 
related to https://github.com/ZcashFoundation/zebra/issues/1287

### CODE REVIEW

**Comment by teor2345** [comment]
(I've removed jvff as a reviewer, so he can focus on network stuff.)

File: zebra-chain/src/orchard/sinsemilla.rs [hunk 1]
```diff
@@ -126,19 +170,68 @@ pub fn sinsemilla_hash(D: &[u8], M: &BitVec<Lsb0, u8>) -> pallas::Base {
...
+    D: &[u8],
+    M: &BitVec<Lsb0, u8>,
+) -> Option<pallas::Point> {
+    incomplete_addition(
+        sinsemilla_hash_to_point(&[D, b"-M"].concat(), M),
+        Some(pallas_group_hash(&[D, b"-r"].concat(), b"") * r),
+    )
 }
 
 /// SinsemillaShortCommit_r(D, M) := Extract_P(SinsemillaCommit_r(D, M))
```
Comment (conradoplg) [inline zebra-chain/src/orchard/sinsemilla.rs]:
```suggestion
/// SinsemillaShortCommit_r(D, M) := Extract⊥ P(SinsemillaCommit_r(D, M))
```

**Review by conradoplg — CHANGES_REQUESTED** [review]
Looks good! :+1: 

There is just a small documentation fix

File: zebra-chain/src/orchard/sinsemilla.rs (hunk 1 above)
Comment (dconnolly) [reply zebra-chain/src/orchard/sinsemilla.rs]: Good catch, thanks!

**Review by conradoplg — APPROVED** [review]

**Pull request merged**